webapp/Dockerfile

```diff
@@ -1,14 +1,16 @@
 FROM node:14-alpine
 
-RUN apk add --update npm
+# https://nodejs.org/docs/latest-v14.x/api/corepack.html#corepack_enabling_the_feature
+RUN corepack enable
 
 WORKDIR /webapp
 
-COPY . .
+# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy
+COPY ["package.json", "yarn.lock", "/webapp"]
+
+RUN yarn install
 
-RUN cd /webapp && \ 
```

Comment:
so this way we won't have to reinstall deps when we change the src (lol). much better